spanner/tests/system.py

```diff
@@ -228,16 +247,49 @@ def test_update_database_ddl(self):
 
         self.assertEqual(len(temp_db.ddl_statements), len(DDL_STATEMENTS))
 
+    def test_db_batch_insert_then_db_snapshot_read_and_db_read(self):
+        from google.cloud.spanner import KeySet
+        keyset = KeySet(all_=True)
 
-class TestSessionAPI(unittest.TestCase):
-    TABLE = 'contacts'
-    COLUMNS = ('contact_id', 'first_name', 'last_name', 'email')
-    ROW_DATA = (
-        (1, u'Phred', u'Phlyntstone', u'phred@example.com'),
-        (2, u'Bharney', u'Rhubble', u'bharney@example.com'),
-        (3, u'Wylma', u'Phlyntstone', u'wylma@example.com'),
-    )
-    SQL = 'SELECT * FROM contacts ORDER BY contact_id'
+        retry = RetryInstanceState(_has_all_ddl)
+        retry(self._db.reload)()
+
+        with self._db.batch() as batch:
+            batch.delete(self.TABLE, keyset)
+            batch.insert(self.TABLE, self.COLUMNS, self.ROW_DATA)
+
+        with self._db.snapshot(read_timestamp=batch.committed) as snapshot:
+            from_snap = list(snapshot.read( self.TABLE, self.COLUMNS, keyset))
+
+        self._check_row_data(from_snap)
+
+        from_db = list(self._db.read(self.TABLE, self.COLUMNS, keyset))
+        self._check_row_data(from_db)
+
+    def test_db_run_in_transaction_then_db_execute_sql(self):
+        from google.cloud.spanner import KeySet
+        keyset = KeySet(all_=True)
+
+        retry = RetryInstanceState(_has_all_ddl)
+        retry(self._db.reload)()
+
+        with self._db.batch() as batch:
+            batch.delete(self.TABLE, keyset)
+
+        def _unit_of_work(transaction, test):
```

Comment:
Why?  The "normal" use of  `run_in_transaction` is to pass in a function with additional varargs / kwargs as needed, which is exactly what we are doing here.